src/com/etendoerp/dependencymanager/datasource/AddDependecyDS.java

```diff
@@ -1,42 +1,363 @@
+          Collectors.toList());
+    }
+    sortResult(parameters, result);
     return result;
   }
+
+  /**
+   * Sorts the given data based on the sorting criteria provided in the parameters.
+   *
+   * @param parameters
+   *     A map containing the sorting criteria.
+   * @param result
+   *     A list of maps representing the data to be sorted.
+   */
+  private void sortResult(Map<String, String> parameters, List<Map<String, Object>> result) {
+    String strSortBy = parameters.getOrDefault(SORT_BY, LINE);
+    Collections.sort(result, new ResultComparator(strSortBy));
+  }
+
+  /**
+   * Reads and processes filtering criteria from the provided parameters to create a `DependencySelectedFilters` object.
+   *
+   * @param parameters
+   *     A map containing the filtering criteria in JSON format.
+   * @return A `DependencySelectedFilters` object populated with the extracted criteria.
+   * @throws JSONException
+   *     if there is an error parsing the JSON data.
+   */
+  private DependencySelectedFilters readCriteria(Map<String, String> parameters) throws JSONException {
+    DependencySelectedFilters selectedFilters = new DependencySelectedFilters();
+    JSONArray criteriaArray = (JSONArray) JsonUtils.buildCriteria(parameters).get(CRITERIA);
+
+    for (int i = 0; i < criteriaArray.length(); i++) {
+      JSONObject criteria = criteriaArray.getJSONObject(i);
+      if (criteria.has(CONSTRUCTOR) && StringUtils.equals(ADVANCED_CRITERIA,
+          criteria.getString(CONSTRUCTOR)) && criteria.has(CRITERIA)) {
+        JSONArray innerCriteriaArray = new JSONArray(criteria.getString(CRITERIA));
+        for (int j = 0; j < innerCriteriaArray.length(); j++) {
+          criteria = innerCriteriaArray.getJSONObject(j);
+          addCriteria(selectedFilters, criteria);
+        }
+      } else {
+        addCriteria(selectedFilters, criteria);
+      }
+    }
+    return selectedFilters;
+  }
+
+  /**
+   * Adds filtering criteria from a JSON object to the specified `DependencySelectedFilters` object.
+   *
+   * @param selectedFilters
+   *     The `DependencySelectedFilters` object to which the criteria should be added.
+   * @param criteria
+   *     A JSON object containing the filtering criteria.
+   * @throws JSONException
+   *     if there is an error parsing the JSON object.
+   */
+  private void addCriteria(DependencySelectedFilters selectedFilters, JSONObject criteria) throws JSONException {
+    String fieldName = criteria.getString(FIELD_NAME);
+    String value = StringUtils.EMPTY;
+    if (criteria.has(VALUE)) {
+      value = criteria.getString(VALUE);
+    }
+
+    if (StringUtils.equals(fieldName, GROUP)) {
+      selectedFilters.setGroup(value);
+    }
+    if (StringUtils.equals(fieldName, ARTIFACT)) {
+      selectedFilters.setArtifact(value);
+    }
+    if (StringUtils.equals(fieldName, VERSION)) {
+      selectedFilters.setVersion(value);
+    }
```

Comment:
Some questions:

If criteria.has(VALUE) is false, does it make sense to execute the following instructions?

Also, it seems that fieldName can only contain one value, that said I think the instructions should contain a return since when one condition is met the others stop making sense. Something like below:

```
  private void addCriteria(DependencySelectedFilters selectedFilters, JSONObject criteria) throws JSONException {
    String fieldName = criteria.getString(FIELD_NAME);
    String value = StringUtils.EMPTY;
    if (criteria.has(VALUE)) {
      value = criteria.getString(VALUE);
    }

    if (StringUtils.equals(fieldName, GROUP)) {
      selectedFilters.setGroup(value);
      return;
    }
    if (StringUtils.equals(fieldName, ARTIFACT)) {
      selectedFilters.setArtifact(value);
      return;
    }
    if (StringUtils.equals(fieldName, VERSION)) {
      selectedFilters.setVersion(value);
      return;
    }
}
```

Reply:
You're right, adding a return makes sense both when criteria doesn't have VALUE and when the fieldName matches. Thanks!